product_supplierinfo_for_customer/__manifest__.py

```diff
@@ -13,7 +13,7 @@
     "category": "Sales Management",
     "license": "AGPL-3",
     "maintainers": ["luisg123v"],
-    "depends": ["product"],
+    "depends": ["sale"],
```

Comment:
You are adding this on wrong module. It should be added on this one: https://github.com/OCA/sale-workflow/blob/17.0/product_supplierinfo_for_customer_sale/__manifest__.py 

That one must not have `sale` module dependency.